src/templates/slideshow/slideshow.js

```diff
@@ -0,0 +1,116 @@
+import { React, useState, useRef, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import styled, { keyframes } from 'styled-components';
+import './slideshow.scss';
+
+function Slideshow({ content }) {
+  const { images, transitions, animations, logo } = content;
+  const logoClasses = `logo ${logo.position} ${logo.size}`;
+  const [index, setIndex] = useState(0);
+  const [Slide, setSlide] = useState();
+  const timeoutRef = useRef(null);
+  const classes = `image ${transitions} ${animations}`;
+
+  // A random function to simplify the code where random is used
+  function random(multiplier) {
+    return Math.floor(Math.random() * multiplier);
+  }
+
+  // Creates the animation using keyframes from styled components
+  function createAnimation(grow, transform) {
+    const transformOrigin = transform || '50% 50%';
+    const startSize = grow ? 1 : 1.2;
+    const finishSize = grow ? 1.2 : 1;
+    const startFinishOpacity = transitions === 'fade' ? 0 : 1;
+    return keyframes`
+    0% {
+      transform: scale(${startSize});
+      transform-origin: ${transformOrigin};
+      opacity: ${startFinishOpacity};
+    }
+    5% {
+      transform: scale(${startSize});
+      transform-origin: ${transformOrigin};
+      opacity: 1;
+    }
+    95% {
+      transform: scale(${finishSize});
+      transform-origin: ${transformOrigin};
+      opacity: 1;
+    }
+    100% {
+      transform: scale(${finishSize});
+      transform-origin: ${transformOrigin};
+      opacity: ${startFinishOpacity};
+    }
+  `;
+  }
+
+  // Determines which animation should be used
```

Comment:
Functions should a 
/** */ comment with parameter descriptions instead of line-comment.